WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -349,7 +367,10 @@ private extension PointOfSaleAggregateModel {
                 onOnboardingCancellation = onCancel
                 return viewModel
             }
-            .assign(to: &$cardPresentPaymentOnboardingViewModel)
+            .sink(receiveValue: { [weak self] onboardingViewModel in
```

Comment:
Do you think some of the memory leak issues could get resolved this way? I treated `assign` as a syntaxic sugar to `sink` but some of our memory debugger observations are pointing to these assignments 🤔

Reply:
Possibly... but probably not. `assign(to: on:)` keeps a strong reference though, perhaps it's worth looking for that deeper in the codebase?

> If you instead implemented MyModel with assign(to: lastUpdated, on: self), storing the returned [AnyCancellable](doc://com.apple.documentation/documentation/combine/anycancellable) instance could cause a reference cycle, because the [Subscribers.Assign](doc://com.apple.documentation/documentation/combine/subscribers/assign) subscriber would hold a strong reference to self. Using assign(to:) solves this problem.
> – [`assign(to:)` docs](https://developer.apple.com/documentation/combine/anypublisher/assign(to:))

Reply:
@staskus Replacing the publishers and associated `sink` calls with simple properties might help (see my other comment) but it probably won't be practical to break it for the Payments code, because that's used right down at the `StripeCardReaderService` level.